internal/backend/invite_request.go

```diff
@@ -36,6 +36,18 @@ func (d *delphisBackend) RequestAccessToDiscussion(ctx context.Context, userID,
 		Status:       model.InviteRequestStatusPending,
 	}
 
+	// Check if user already has access
+	dua, err := d.GetDiscussionUserAccess(ctx, discussionID, userID)
+	if err != nil {
+		logrus.WithError(err).Error("failed to get discussion user access")
+		return nil, err
+	}
+
+	if dua != nil && dua.State == model.DiscussionUserAccessStateActive {
+		logrus.Infof("user already has access ")
+		return nil, nil
```

Comment:
do we want to return nil or an error?

Reply:
I think it's reasonable to return the request instead of nil?